### PR TITLE
Fix typo of rate limiter

### PR DESCRIPTION
**Pull request by astraw99**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Fix the doc typo to better understand `RateLimiter`.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 type RateLimiter interface {
 	// When gets an item and gets to decide how long that item should wait
 	When(item interface{}) time.Duration
-	// Forget indicates that an item is finished being retried.  Doesn't matter whether its for perm failing
```
Comment (yliaog) [inline staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters.go:30]: i think perm is for 'permanent'

Comment (astraw99) [reply staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters.go:30]: Fixed, PTAL thanks.

**Comment by astraw99** [comment]
@yliaog PTAL, thanks.

**Pull request merged**